cmd/ingester/app/consumer/rate_limiter.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumer
+
+import (
+	"sync"
+	"time"
+)
+
+// rateLimiter wraps a channel to provide rate limiting.
+type rateLimiter struct {
```

Comment:
IMHO, this rate limiter seems pretty complex in that it requires a bunch of channels, a wait group, and a Ticker.

Seeing that this rate limiter is only being used from a single threaded context, and that we are most interested in the stable behavior of the rate limiter,  what advantages does it have over an simpler implementation like one described in https://stackoverflow.com/questions/667508/whats-a-good-rate-limiting-algorithm ?

Reply:
Done.